.pre-commit-config.yaml

```diff
@@ -0,0 +1,44 @@
+exclude: "docs|.git|geometric_data|geometric_features/features_and_tags.json"
+default_stages: [pre-commit]
+fail_fast: true
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+
+  # Can run individually with `pre-commit run isort --all-files`
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.13.2
+    hooks:
+      - id: isort
+        exclude: "geometric_features/__init__.py"
+
+  # Can run individually with `flynt [file]` or `flynt [source]`
+  - repo: https://github.com/ikamensh/flynt
+    rev: '1.0.1'
+    hooks:
+      - id: flynt
+        args: ["--fail-on-change", "--verbose"]
+        require_serial: true
+
+  # Can run individually with `pre-commit run flake8 --all-files`
+  # Need to use flake8 GitHub mirror due to CentOS git issue with GitLab
+  # https://github.com/pre-commit/pre-commit/issues/1206
+  - repo: https://github.com/pycqa/flake8
+    rev: 7.1.1
+    hooks:
+      - id: flake8
+        args: ["--config=.flake8"]
+        additional_dependencies: [flake8-isort]
+
+  # Can run individually with `pre-commit run mypy --all-files`
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.13.0
+    hooks:
+      - id: mypy
+        args: ["--config=pyproject.toml", "--show-error-codes"]
+        verbose: true
+        additional_dependencies: ['types-requests', 'types-PyYAML']
```

Comment:
I think we had talked about this in our meeting but I would rather leave `mypy` out on this repo.  In my experience, it's a step too far unless you start with it from the beginning.